reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -665,10 +726,16 @@ def binary_mask_3d(center, radius, shape, fill=True, shell_thickness=1):
             An array of bools with "True" where the sphere mask is
             present.
         """
-        a, b, c = center
-        nx0, nx1, nx2 = shape
-        x0, x1, x2 = np.ogrid[-a : nx0 - a, -b : nx1 - b, -c : nx2 - c]
-        r2 = x0**2 + x1**2 + x2**2
+        if d == 3:
+            a, b, c = center
+            nx0, nx1, nx2 = shape
+            x0, x1, x2 = np.ogrid[-a : nx0 - a, -b : nx1 - b, -c : nx2 - c]
+            r2 = x0**2 + x1**2 + x2**2
+        else:
```

Comment:
I would change to `elif d==2` and have else throw a "not implemented" error. The github workflows will complain if you have an assert in this module (because it's not a test), so throw an error some other way.